### PR TITLE
Fix: Listing projects comprising of only the user's on listAll=true

### PR DESCRIPTION
**Pull request by Pearl1594**
## Description
1. A few changes made to `server/src/main/java/com/cloud/api/query/QueryManagerImpl.java` were reverted by commit: `fc05d3168f14584431a1a6b3c1f88cf04999cfaa`
2. Fix a check for parameters passed

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]

**Comment by Pearl1594** [comment]
@blueorangutan package

File: api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java [hunk 1]
```diff
@@ -116,7 +116,7 @@ public long getEntityOwnerId() {
             throw new InvalidParameterValueException("Account name and domain id must be specified together");
         }
 
-        if (userId != null && (accountId == null && domainId == null)) {
+        if (userId != null && (accountId == null || domainId == null)) {
```
Comment (DaanHoogland) [inline api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java:119]: why should they both be specified? user + either implies the other

Comment (DaanHoogland) [reply api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java:119]: ignore, bad reading on my part

Comment (Pearl1594) [reply api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java:119]: @DaanHoogland this was the requirement in the FR doc

**Comment by Pearl1594** [comment]
@blueorangutan test

File: api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java (hunk 1 above)
Comment (DaanHoogland) [reply api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java:119]: discussed off-line with @Pearl1594 . no functional need but bugs may arise when not implemented this way

**Review by harikrishna-patnala — APPROVED** [review]
LGTM, test failure is not related to these changes.

**Comment by DaanHoogland** [comment]
travis has more errors than expected, retrying

**Comment by DaanHoogland** [comment]
this and #4276 look like they are related!?

File: api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java [hunk 2]
```diff
@@ -116,7 +116,7 @@ public long getEntityOwnerId() {
...
             throw new InvalidParameterValueException("Account ID and Domain ID must be specified with userID");
```
Comment (sureshanaparti) [inline api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java:120]: @Pearl1594 update this msg to reflect the change in cond.

Comment (DaanHoogland) [reply api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java:120]: It already did, as far as I can tell, neither accountId nor domainId may be null so both must be specified.

Comment (sureshanaparti) [reply api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java:120]: am i missing anything here @DaanHoogland , the cond. is OR and the msg says both (account id and domain id).

Comment (DaanHoogland) [reply api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java:120]: throw an exception when either is null, would mean both may not be null. The message was already the (not yet implemented) new condition.

Comment (sureshanaparti) [reply api/src/main/java/org/apache/cloudstack/api/command/user/project/CreateProjectCmd.java:120]: hmmm , correct @DaanHoogland

**Review by sureshanaparti — APPROVED** [review]

**Comment by DaanHoogland** [comment]
no test effort shows in the comments or description @Pearl1594 . Can you add or chase some?
if you feel this is covered by smoke tests, please mention those here.
travis is lacking I think we can ignore it for this one.

**Comment by Pearl1594** [comment]
> no test effort shows in the comments or description @Pearl1594 . Can you add or chase some?
> if you feel this is covered by smoke tests, please mention those here.
> travis is lacking I think we can ignore it for this one.

@DaanHoogland Tests performed were listed  in the following PR: https://github.com/apache/cloudstack/pull/4316
I've just raised this PR as it was reverted while merging some code.

**Comment by DaanHoogland** [comment]
So are the same `before:` and `after:` entries applicable here @Pearl1594 ?

**Comment by Pearl1594** [comment]
> So are the same `before:` and `after:` entries applicable here @Pearl1594 ?

yes @DaanHoogland

**Comment by weizhouapache** [comment]
what about 4.14 ?

**Comment by DaanHoogland** [comment]
@weizhouapache this fixes a regression caused in a 4.15 commit on master. the original was also targeted on 4.15. not sure if it is applicable to 4.14. If it is it must be back-ported.

**Comment by Pearl1594** [comment]
@DaanHoogland @weizhouapache  this fix was specific to a feature introduced in 4.15.

**Comment by weizhouapache** [comment]
@DaanHoogland @Pearl1594 thanks, clear
it is a regression of #4307

**Pull request merged**